no-downtime-upgrade/src/main/java/org/keycloak/playground/nodowntimeupgrade/base/storage/Storage.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.playground.nodowntimeupgrade.base.storage;
+
+import org.keycloak.playground.nodowntimeupgrade.base.model.HasId;
+import java.util.Set;
+
+/**
+ *
+ * @author hmlnarik
+ */
+public interface Storage<ModelType extends HasId<Integer>> {
+    /**
+     * Creates an object in the store. Introduces new ID.
+     * @param object
+     * @return ID of the object
+     */
+    Integer create(ModelType object);
+
+    /**
+     * Returns object with the given {@code id} from the database or {@code null} if object does not exist.
```

Comment:
Nitpick: The javadoc talks about "database", but it is probably desired to use the word "storage" to be compatible with "create" method? Same comment applies to other methods in this interface.